src/mono/mono/mini/method-to-ir.c

```diff
@@ -7585,7 +7585,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 
 				// Create a new dummy signature with the lowered arguments				
-				fsig = mono_metadata_signature_dup_new_params (cfg->mempool, fsig, new_param_count, (MonoType**)new_params->data);
+				fsig = mono_metadata_signature_dup_new_params (cfg->mem_manager, fsig, new_param_count, (MonoType**)new_params->data);
```

Comment:
Is this really needed though ? While interpreter requires the signature to be alive after the code is compiled, I don't think jit code has this constraint. This change prevents the signature from being freed as it was done before.

Reply:
We could use similar approach as with `mono_metadata_signature_dup_internal` https://github.com/dotnet/runtime/blob/b1b3e85dbf2d52b70d8c44c663e83b5c320c6abd/src/mono/mono/metadata/metadata.c#L2429 and use `mempool` for JIT and `mem_manager` for interpreter.

Reply:
Thanks, updated.